library/Icinga/File/Pdf.php

```diff
@@ -61,12 +62,29 @@ public function renderControllerAction($controller)
             'src="' . Icinga::app()->getBootstrapDirectory() . '/img/',
             $html
         );
+
+        $request = $controller->getRequest();
+
+        if (Hook::has('Pdfexport')) {
+            $pdfexport = Hook::first('Pdfexport');
+
+            if ($pdfexport->isSupported()) {
+                $pdfexport->streamPdfFromHtml($html, sprintf(
```

Comment:
What about moving this to a dedicated `PdfExport` class? `File/Pdf` has similar logic right now and IMO that's also not the right place. But this Hook-based logic is one more logical component you wouldn't expect here.

Reply:
You already said that `File/Pdf` is already full of logic which you would not expect there. So, why bother?

Reply:
Because it is a Hook, it's per definition not internal. If you want it to be internal, do not create a Hook ;-)

Reply:
I don't get it. `File\Pdf` is a useless library class for others. It would make sense to rewrite all that stuff, sure. The hook could introduce a new method which just renders HTML to PDF though. The current implementation is for Web 2's PDF export only. And there's no need for more at the moment.